lib/AbstractFlatOperation.ts

```diff
@@ -14,23 +14,25 @@ export abstract class AbstractFlatOperation<LoadedValue> extends AbstractOperati
     const loadingPromise = this.resolveValue(key)
     this.runningLoads.set(key, loadingPromise)
 
-    const resolvedValue = await loadingPromise
-    if (resolvedValue === undefined) {
-      if (this.throwIfUnresolved) {
-        this.runningLoads.delete(key)
-        throw new Error(`Failed to resolve value for key "${key}"`)
+    loadingPromise.then((resolvedValue) => {
+      if (resolvedValue === undefined) {
+        if (this.throwIfUnresolved) {
+          this.runningLoads.delete(key)
+          throw new Error(`Failed to resolve value for key "${key}"`)
```

Comment:
wouldn't this error end up being an unhandledexception and ultimately crash an application, since there is no way anything above can handle it? I would suggest removing this error throwing altogether then

Reply:
Moved error logic to value resolvers